### PR TITLE
[9.x.x] Bugfix 1209255: Issue with reflections when using an orthographic camera

### PR DESCRIPTION
**Pull request by ellioman**
## Important info for backporting
These PRs must be backported with this one:
- ShaderGraph Preview change revert #358
- Android Reference Images update #355

## Purpose of this PR
When using orthographic cameras, the reflection would be incorrect because the view direction was calculated the same way for both perspective & orthographic cameras.

https://issuetracker.unity3d.com/issues/urp-weird-reflections-when-using-lit-material-and-a-camera-with-orthographic-projection

This PR does the following:
- Adds GetWorldSpaceViewDir() to ShaderVariablesFunctions.hlsl
- Adds #include for the ShaderVariablesFunctions file and calls that function to calculate the viewDirWS in Lit, SimpleLit, ParticlesLit, ParticlesSimpleLit Varyings, PreviewVaryings, SpeedTree7Passes, SpeedTree7BillboardPasses, SpeedTree7Passes, and TerrainLit ForwardPass hlsl files.
- Updates some reference images for scenes that use orthographic cameras

---

### Checklist for PR maker
- [x] Have you added a backport label (if needed)? For example, the `need-backport-2019.3` label. After you backport the PR, the label changes to `backported-2019.3`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR.
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---

### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Yamato**: 
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fbugfix-1209255-ortho-reflections

### CODE REVIEW

**Comment by LandonTownsendUnity** [comment]
Needs to be fixed for shadergraphs as well:
image
Please fix for both PBR and Unlit graphs as well as making sure that the View Direction node uses the new function in all spaces so that it is correct in orthographic mode.

**Review by ghost — APPROVED** [review]
View Dir confirmed fixed for both PBR and Unlit in World space, Object space and View space.
Reflections for PBR shadergraph shaders confirmed fixed in orthographic.
Behavior for view dir with non-orthographic consistent with previous results.

**Review by alindmanUnity — APPROVED** [review]
shader graph code lgtm

**Comment by phi-lira** [comment]
This PR is adding .lighting files as well. These files are only 2020.1+, be careful when backporting to 7.x

File: com.unity.shadergraph/ShaderGraphLibrary/ShaderVariablesFunctions.hlsl [hunk 1]
```diff
@@ -45,19 +45,14 @@ bool IsPerspectiveProjection()
 }
 
 // Computes the world space view direction (pointing towards the viewer).
+float3 GetWorldSpaceViewDir(float3 positionWS)
```
Comment (phi-lira) [inline com.unity.shadergraph/ShaderGraphLibrary/ShaderVariablesFunctions.hlsl:48]: This is not ideal. Consider we want to have a custom hlsl shader for URP, now we have to include shader graph files, this should be in URP and shader graph include it.

**Review by phi-lira — APPROVED** [review]

**Comment by phi-lira** [comment]
Adding @PaulDemeulenaere for awareness in changes in VFX Graph. For context, the motivation is to make these core functions in the pipeline so we can also use in custom hlsl shaders. This is similar to what HDRP has.

**Review by PaulDemeulenaere — APPROVED** [review]
I had some doubt about shadow pass in VFX (because SHADERPASS_SHADOWS isn't defined) but it doesn't change against the previous behavior.
LGTM!
Thanks !

**Comment by ellioman** [comment]
@PaulDemeulenaere Hey, 
Actually I just made a commit which actually does change the behaviour.
Since we don't support camera relative rendering me and @phi-lira decided to comment out that piece of code with a comment.

So now when you call `GetCurrentViewPosition()` we will return `_WorldSpaceCameraPos` instead of `UNITY_MATRIX_I_V._14_24_34`

Do you see any issues with that? All the tests on Yamato are green.

**Comment by ellioman** [comment]
Yamato was green before I did a changelog and buildsetting commit to fix conflicts. I therefore cancelled the Yamato run.

**Pull request merged**